test/test_main.py

```diff
@@ -11,8 +12,36 @@
     sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))
     from main import run
 
-test_files = glob.glob(os.path.join(os.path.dirname(__file__), "files/**/*.ifc"), recursive=True)
-@pytest.mark.parametrize("filename", test_files)
+rule_code_pattern = re.compile(r"^[a-zA-Z]{3}\d{3}$")
```

Comment:
You have `rule_code_pattern` and `rule_codes` duplicated in the function body on line 27.